appcues/src/main/java/com/appcues/statemachine/StateMachineFactory.kt

```diff
@@ -0,0 +1,17 @@
+package com.appcues.statemachine
+
+import com.appcues.AppcuesConfig
+import com.appcues.AppcuesCoroutineScope
+import com.appcues.action.ActionProcessor
+import com.appcues.analytics.ExperienceLifecycleTracker
+import com.appcues.statemachine.State.Idling
+
+internal class StateMachineFactory(
```

Comment:
instead of relying on Koin its much more readable that we have a factory that produces state machines.